packages/ui/src/components/va-tree-view/types.ts

```diff
@@ -0,0 +1,51 @@
+import { ComputedRef, InjectionKey, Ref } from 'vue'
+
+export interface TreeNode {
+  id: number | string
+  children: TreeNode[]
+  level?: number
+  checked?: boolean
+  disabled?: boolean
+  expanded?: boolean
+  hasChildren?: boolean
+  matchesFilter?: boolean
+  [key: string]: any
+}
+
+export interface TreeView {
+  nodeKey: string | number
+  labelKey: string
+  selectable: boolean
+  iconColor: ComputedRef<string>
+  colorComputed: ComputedRef<string>
+  toggleNode: (node: TreeNode) => void
+  toggleCheckbox: (node: TreeNode, isSelected: boolean) => void
+  getKey: (node:TreeNode) => string,
+}
+
+export const TreeViewKey = Symbol('TreeView') as InjectionKey<TreeView>
+
+export type CreateNodeProps = {
+  node: TreeNode
+  level?: number
+  children?: TreeNode[]
+  filter?: Ref<string>
+  labelKey?: Ref<string>
+  expandAll?: Ref<boolean>
+}
+
+export type CreateNodeFunc = (props: CreateNodeProps) => TreeNode
+
+export type UseTreeBuilderProps = {
```

Comment:
https://github.com/epicmaxco/vuestic-ui/issues/1907#issuecomment-1195712961

types.ts must include only types, that required for user. Store inner types like UseTreeBuilderProps in composables.